### ISSUE TITLE
Some stuff to fix from pvp players feedback

### ISSUE THREAD

**Issue by dydzio0614** [vcmi/vcmi#4109]
- when the timer runs out in a fight, the hero runs away, it should be possible to choose between retreat and finish with an autocombat
- if a shooter can cast a spell (e.g. elementals) he cannot pick melee attack from alternative actions
- ruins (hota highlands object) do not reveal the terrain
- view earth shows where heroes are
- game loading: turn settings UI is not synced with save data
- backpack for artifact does not react to mouse scroll
- GUI arrows under creature slots in exchange window should work differently with alt/ctrl button pressed (moving result to another hero)
- swap functions of ctrl+click and shift+click on adventure map
- cannot quit game via UI while other player is making turn, that also means host will be stuck without alt+f4 if game has AI's and non-host quits while making his turn

**Comment by IvanSavenko** [comment]
1. Probably not trivial at the moment. Currently the only alternative is "end turn once combat is over"
2. May take a look if I have time/interest, but no promises - vcmi extras is a mess.
3. Not implemented
9. May take a look, not sure if a quick fix or something more complex, so may not be for 1.5.3

Rest - will take a look, sounds simple enough.

**Comment by dydzio0614** [comment]
7. may be hard due to VCMI extras as well

**Comment by dydzio0614** [comment]
the way i personally see target implementation of "1": blocking yes/no popup appears where "yes" finishes battle with autocombat (functionality now exists), and "no" button triggers retreat

**Comment by IvanSavenko** [comment]
4 - Unable to reproduce. Might be caused by h3 bug #1441 (fixed in hota)
7 - Postponing, unless somebody else looks into it. It requires more than just code - tooltip to indicate change in function, new icons to indicate it, possibly more.

6 & 8 - fixed, will open PR soon

**Comment by krs0** [comment]
Yup, run into 4 as well. You can actually see castles and heroes :)

![image](https://github.com/user-attachments/assets/149864d1-600e-4252-9294-56165a67bbed)
